openedx/openedx-platform
# feat: add page redirect when mfe flag enabled

## Description
This PR adds a page redirect to the Course Authoring MFE when its associated waffle flag is enabled. For example, when the `contentstore.new_studio_mfe.use_new_custom_pages` is enabled and a user tries to navigate to `{studioEndpoint}/tabs/{course_key}`, they will be redirected to {courseAuthoringEndpoint}/course/{course_key}/custom-pages`. This change impacts Authors.

## Supporting information

JIRA Ticket: [TNL-10780](https://2u-internal.atlassian.net/browse/TNL-10780)

## Testing instructions

1. Navigate to a course outline.
2. In a separate tab, open Django Admin and enable the `contentstore.new_studio_mfe.use_new_course_outline_page` waffle flag.
3. Go back to the course outline and reload the page.
4. Address bar should no longer be `localhost:18010`, but `localhost:2001`.
5. Turn of the waffle flag.
6. Reload the course outline page.
7. The address bar should still be `localhost:2001`.
8. Change 2001 to 18010 and the legacy studio page should render.

## Deadline

None
